### ISSUE TITLE
It seems to be a bug！！！

### ISSUE THREAD

**Issue by shilida** [pytorch/pytorch#67835]
code1:
`            for step, (arg, cxt) in enumerate(tqdm_obj):
                batch = [tuple(t.to(self.device) for t in arg), tuple(t.to(self.device) for t in cxt)]
                output = []
                hidden_list = []
                for i in range(2):
                    if i == 0:
                        logits_0,hidden_0 = self.model[0](*batch[0][:-1])
                        output.append(logits_0)
                        hidden_list.append(hidden_0)
                    else:
                        logits_1, hidden_1 = self.model[1](*batch[1][:-1])
                        output.append(logits_1)
                        hidden_list.append(hidden_1)
                loss = self.criterion(output[0], batch[0][-1])
                ......
 `
code2:
`            for step, (arg, cxt) in enumerate(tqdm_obj):
                batch = [tuple(t.to(self.device) for t in arg), tuple(t.to(self.device) for t in cxt)]
                output = []
                hidden_list = []
                for i in range(2):
                    if i == 0:
                        logits_0,hidden_0 = self.model[0](*batch[0][:-1])
                        output.append(logits_0)
                        hidden_list.append(hidden_0)
                    # else:
                    #     logits_1, hidden_1 = self.model[1](*batch[1][:-1])
                    #     output.append(logits_1)
                    #     hidden_list.append(hidden_1)
                loss = self.criterion(output[0], batch[0][-1])
`
These two codes will get different results only because code1 computes this extra part：
`
                        logits_1, hidden_1 = self.model[1](*batch[1][:-1])
                        output.append(logits_1)
                        hidden_list.append(hidden_1)
`
 But this part is not used ！！！！
My version of pytorch is 1.7.1 and GPU is TITAN RTX.

**Comment by soulitzer** [comment]
Hey thanks for the report. It's hard to say what is wrong from the information here. There could be shared state between your models, or there might be some source of randomness you aren't controlling for. 

You can also try reproducing the issue again on a later version of PyTorch like 1.10.
